bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/ApiBridgeHandler.java

```diff
@@ -0,0 +1,240 @@
+import org.openhab.binding.netatmo.internal.discovery.NetatmoDiscoveryService;
+import org.openhab.binding.netatmo.internal.webhook.NetatmoServlet;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link ApiBridgeHandler} is the handler for a Netatmo API and connects it to the framework.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class ApiBridgeHandler extends BaseBridgeHandler {
+    private static final int TIMEOUT_S = 20;
+
+    private final Logger logger = LoggerFactory.getLogger(ApiBridgeHandler.class);
+    private final BindingConfiguration bindingConf;
+    private final HttpService httpService;
+    private final AuthenticationApi connectApi;
+    private final HttpClient httpClient;
+    private final NADeserializer deserializer;
+
+    private Optional<ScheduledFuture<?>> connectJob = Optional.empty();
+    private Optional<NetatmoServlet> servlet = Optional.empty();
+    private @NonNullByDefault({}) ApiHandlerConfiguration thingConf;
+
+    private Map<Class<? extends RestManager>, RestManager> managers = new HashMap<>();
+
+    public ApiBridgeHandler(Bridge bridge, HttpClient httpClient, HttpService httpService, NADeserializer deserializer,
+            BindingConfiguration configuration) {
+        super(bridge);
+        this.bindingConf = configuration;
+        this.httpService = httpService;
+        this.connectApi = new AuthenticationApi(this, scheduler);
+        this.httpClient = httpClient;
+        this.deserializer = deserializer;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Netatmo API bridge handler.");
+        thingConf = getConfigAs(ApiHandlerConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
+            openConnection();
+            String webHookUrl = thingConf.webHookUrl;
+            if (webHookUrl != null && !webHookUrl.isBlank()) {
+                servlet = Optional.of(new NetatmoServlet(httpService, this, webHookUrl));
+            }
+        });
+    }
+
+    private void openConnection() {
+        try {
+            Credentials credentials = thingConf.getCredentials();
+            logger.debug("Connecting to Netatmo API.");
+            try {
+                connectApi.authenticate(credentials, bindingConf.features);
+                updateStatus(ThingStatus.ONLINE);
+                getThing().getThings().stream().filter(Thing::isEnabled).filter(t -> t instanceof CommonInterface)
```

Comment:
t will never be an instance of CommonInterface because t is a `Thing` and not a `ThingHandler`.
That being said, as `expireData` was never run by this code, you should think if it is really necessary or not.

Reply:
You're correct. Now that you've implemented bridge status change cascading, it is no more needed.

Reply:
Thinking about it, yes, this code is useful in case of reconnection.
So please just fix it.

Reply:
Ok, I saw your message after writing.
So if you think it is no more necessary, remove it.

Reply:
Done.